diff --git a/arch/mips/xburst/soc-t23/chip-t23/isvp/common/spi_bus.c b/arch/mips/xburst/soc-t23/chip-t23/isvp/common/spi_bus.c
index cd06b8ac..b8eb434d 100644
--- a/arch/mips/xburst/soc-t23/chip-t23/isvp/common/spi_bus.c
+++ b/arch/mips/xburst/soc-t23/chip-t23/isvp/common/spi_bus.c
@@ -153,6 +153,27 @@ struct spi_nor_platform_data spi_nor_pdata[] = {
 #ifdef CONFIG_SPI_QUAD
 		.quad_mode = &flash_quad_mode[1],
 #endif
+	},
+	{
+		.name           = "EN25QX64A",
+		.pagesize       = 256,
+		.sectorsize     = 4 * 1024,
+		.chipsize       = 8192 * 1024,
+		.erasesize      = 32 * 1024,
+		.id             = 0x1c7117,
+
+		.block_info     = flash_block_info,
+		.num_block_info = ARRAY_SIZE(flash_block_info),
+
+		.addrsize       = 3,
+		.pp_maxbusy     = 3,            /* 3ms */
+		.se_maxbusy     = 400,          /* 400ms */
+		.ce_maxbusy     = 8 * 10000,    /* 80s */
+
+		.st_regnum      = 3,
+#ifdef CONFIG_SPI_QUAD
+		.quad_mode = &flash_quad_mode[1],
+#endif
 	},
 	{
 		.name           = "EN25QH128A",
